include/uxr/agent/middleware/ced/CedMiddleware.hpp

```diff
@@ -28,123 +28,339 @@ class CedMiddleware : public Middleware
+     * @param timeout       The timeout (milliseconds) of the reading.
+     * @return  true in case of successful reading and false in other case.
+     */
     bool read_data(
             uint16_t datareader_id,
             std::vector<uint8_t>& data,
             std::chrono::milliseconds timeout) override;
 
-/**********************************************************************************************************************
- * Matched functions.
- **********************************************************************************************************************/
+    /**
+     * @brief Checks whether an existing CedParticipant, identified by the participant_id, matches with a new
+     *        CedParticipant that would result from the creation of a new one using the domain_id and the reference
+     *        representation.
+     *        It is considered that the CedParticipants match if both have the same domain_id.
+     * @param participant_id    The existing CedParticipant identifier.
+     * @param domain_id         The Domain identifier of the new CedParticipant.
+     * @param ref               The reference that represents the new CedParticipant (unused).
+     * @return  true if both CedParticipant have the same Domain identifier, and false in other case.
+     */
     bool matched_participant_from_ref(
             uint16_t participant_id,
             int16_t domain_id,
             const std::string& ref) const override;
 
+    /**
+     * @brief Checks whether an existing CedParticipant, identified by the participant_id, matches with a new
+     *        CedParticipant that would result from the creation of a new one using the domain_id and the XML
+     *        representation.
+     *        It is considered that the CedParticipants match if both have the same domain_id.
+     * @param participant_id    The existing CedParticipant identifier.
+     * @param domain_id         The Domain identifier of the new CedParticipant.
+     * @param xml               The XML that describes the new CedParticipant (unused).
+     * @return  true if both CedParticipant have the same Domain identifier, and false in other case.
+     */
     bool matched_participant_from_xml(
             uint16_t participant_id,
             int16_t domain_id,
             const std::string& xml) const override;
 
+    /**
+     * @brief Checks whether an existing CedTopic, identified by the topic_id, matches with a new CedTopic that would
+     *        result from the creation of a new one using the reference representation.
+     *        It is considered that the CedTopics match if both have the same topic name.
+     * @param topic_id  The existing CedTopic identifier.
+     * @param ref       The reference that represents the new CedTopic.
+     *                  It defines the CedGlobalTopic name.
+     * @return  true if both CedTopic have associated the same CedGlobalTopic, and false in other case.
```

Comment:
What is CedGlobalTopic?.

Reply:
It is the Topic which is shared between all the Participants, DataWriter and DataReader.